### DISCUSSION TITLE
Android Widget causing phone to crash

### DISCUSSION THREAD

**Discussion by Benjee369** [community/community#143364] [Mobile]
When I try to add the widget that shows your push history to my phone home screen it causes it to crash and restart the system launcher

**Comment by ravindudil5han** [comment]
I'm sorry to hear you're experiencing issues with adding the widget that displays your push history to your phone's home screen. When attempting to add this widget causes your system launcher to crash and restart, it can be quite frustrating. Here's a step-by-step guide to help you resolve this issue:

### **1. Restart Your Phone**
Sometimes, a simple restart can fix temporary glitches.

- **How to Restart:**
  - Press and hold the **Power** button on your device.
  - Select **Restart** from the options that appear.

### **2. Update Your Apps and System Software**
Ensure that both your system software and the app providing the widget are up to date.

- **Update System Software:**
  - Go to **Settings** > **System** > **Software Update**.
  - Check for updates and install if available.

- **Update the Widget App:**
  - Open the **Google Play Store** (Android) or **App Store** (iOS).
  - Search for the app that provides the widget.
  - If an update is available, tap **Update**.

### **3. Clear Launcher Cache (Android Only)**
Clearing the cache of your launcher can resolve issues related to widgets.

- **How to Clear Cache:**
  - Go to **Settings** > **Apps** or **Applications**.
  - Find and select your **Launcher** app (e.g., **Google Pixel Launcher**, **Samsung One UI Home**, etc.).
  - Tap on **Storage**.
  - Select **Clear Cache**.

### **4. Reinstall the Widget App**
There might be an issue with the app itself. Reinstalling it can help.

- **How to Reinstall:**
  - **Uninstall the App:**
    - Press and hold the app icon.
    - Select **Uninstall** or **Remove**.
  - **Reinstall the App:**
    - Go to the **Google Play Store** or **App Store**.
    - Search for the app and install it again.

### **5. Check Widget Compatibility**
Ensure that the widget is compatible with your device and operating system version.

- **How to Check:**
  - Visit the app’s page on the **Google Play Store** or **App Store**.
  - Read the **Description** and **Requirements** to confirm compatibility.

### **6. Try an Alternative Launcher**
If the problem persists, it might be an issue with your current launcher. Trying a different launcher can help identify the problem.

- **Popular Launchers:**
  - **Nova Launcher**
  - **Microsoft Launcher**
  - **Action Launcher**

- **How to Change Launcher:**
  - Install a new launcher from the **Google Play Store** or **App Store**.
  - Press the **Home** button.
  - Select the new launcher as your default.

### **7. Check for Conflicting Apps**
Some apps might interfere with widgets. Identifying and disabling them can resolve the issue.

- **How to Identify:**
  - Think about any recent apps you’ve installed that might interact with your home screen or notifications.
  - Temporarily disable or uninstall these apps to see if the problem persists.

### **8. Contact App Support**
If none of the above solutions work, it might be a bug within the app itself.

- **How to Contact Support:**
  - Visit the app’s page on the **Google Play Store** or **App Store**.
  - Scroll down to **Developer Contact** and reach out via email or their provided support channels.

### **9. Factory Reset (Last Resort)**
If all else fails, performing a factory reset can resolve deep-seated issues. **Note:** This will erase all data on your device, so ensure you back up important information first.

- **How to Factory Reset:**
  - Go to **Settings** > **System** > **Reset Options**.
  - Select **Erase All Data (Factory Reset)**.
  - Follow the on-screen instructions.

### **Summary**
Start with the simpler solutions like restarting your phone and updating your apps. If the issue continues, proceed with clearing the launcher cache, reinstalling the widget app, and checking for compatibility. Trying an alternative launcher or identifying conflicting apps can also help. If none of these steps work, reaching out to the app’s support team or considering a factory reset may be necessary.

I hope one of these solutions helps you resolve the issue with the widget causing your system launcher to crash. If you have any further questions or need additional assistance, feel free to ask!

**Comment by yogeshpaliyal** [comment]
@Benjee369 does it worked?
For me it is happening on mi launcher, it is only happening for Github Widget, all other app widgets are working as expected.


Here is the error stack as well
![Screenshot_2025-01-01-23-34-37-872_com.miui.bugreport.jpg](https://github.com/user-attachments/assets/17a835ac-8b00-4e3d-98a4-fbf13b50c1dd)

**Reply by JameDevOfficial** [reply]
have you found any soltution? Im encountering the same problem

**Reply by yogeshpaliyal** [reply]
No not yet

**Reply by JameDevOfficial** [reply]
Im using an app called "Github widget" as a temporary solution for now. the widget is a bit buggy when scaling it but it does it's job

**Comment by iskae** [comment]
Hey folks 👋 

I am sorry to hear that you are experiencing problems with the widgets on Android. Some of the issues that you have described are known to us, but I would love to test it once again 🙇 

Can you provide me with more details, such as:

- Device details: Name, OS
- Have you sideloaded or installed the app via Play Store?
- The version number for GitHub Mobile App (It can be found in Play Store page)

Thanks in advance!

**Reply by JameDevOfficial** [reply]
Redmi Note 12 Pro+ 5G, HyperOS 2.0
Play Store
1.209.1
